src/main/scala/mesosphere/marathon/core/election/impl/CuratorElectionService.scala

```diff
@@ -67,13 +68,54 @@ class CuratorElectionService(
     maybeLatch.get.start()
   }
 
+  override protected def startLeadership(abdicate: Abdicator): Unit = {
+    maybeLatch match {
+      case None => super.startLeadership(abdicate)
+      case Some(l) =>
+        l.synchronized {
+          super.startLeadership(abdicate)
+        }
+    }
+  }
+
+  override protected def stopLeadership(): Unit = {
+    maybeLatch match {
+      case None => super.stopLeadership
+      case Some(l) =>
+        l.synchronized {
```

Comment:
So, when we are shutting down, what tries to access the latch and why would that lead to a deadlock?

It seems that `triggerShutdown` should be idempotent as @timcharper does [here](https://github.com/mesosphere/marathon/compare/tharper/network-partition#diff-22d2b7f3a6de5a4a45e90407f600ecbdR60) with an atomic boolean.

 Same should be true for `stopLeadership`.  What do you think, @brad-peters?